rpc: add channel reserve fields in RPC responses.

roeierez opened the pull request:
This PR adds two fields to channels that return as part of RPC result endpoint. For both pending and open channels the fields: local_chan_reserve, remote_chan_reserve were added.
These fields represents the reservation the nodes are required to keep in both sides of the channel.
This is provides an accurate way to calculate the "real" inbound and outbound liquidity.
If I understood correctly, I believe it is also a part of the solution to this bug in lightningapp: https://github.com/lightninglabs/lightning-app/issues/1071

Fixes #1639.

Roasbeef commented:
Duplicate of #1998?

roeierez commented:
> Duplicate of #1998?

Both indeed adds the local channel reserve field but this PR adds the remote channel reserve as well.
To me it makes sense to add both fields together.

cfromknecht commented on lnrpc/rpc.proto:
```diff
@@ -1161,6 +1161,13 @@ message Channel {
     string chan_status_flags = 19 [json_name = "chan_status_flags"];
+
+    /// This minimum satoshies this node is required to reseve in its balance    
+    int64 local_chan_reserve = 20 [json_name = "local_chan_reserve"];
```
both fields should have a `_sat` suffix to indicate the unit, same for pending channels response

nit: reseve -> reserve, also add period :)

roeierez replied:
Done.

cfromknecht commented on lnrpc/rpc.proto:
```diff
@@ -1468,6 +1475,9 @@ message PendingChannelsResponse {
         int64 local_balance = 4 [ json_name = "local_balance" ];
         int64 remote_balance = 5 [ json_name = "remote_balance" ];
+       
+        int64 local_chan_reserve = 6 [json_name = "local_chan_reserve"];
```
can also copy the new comments to these two fields. the others _should_ be populated, but we can at least start by making incremental progress

roeierez replied:
Done.

cfromknecht commented on rpcserver.go:
```diff
@@ -29,7 +29,7 @@ import (
 	"github.com/coreos/bbolt"
 	"github.com/davecgh/go-spew/spew"
-	"github.com/grpc-ecosystem/go-grpc-middleware"
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
```
doesn't look like this change is needed

roeierez replied:
Right, removed.

halseth commented on lnrpc/rpc.proto:
```diff
@@ -1161,6 +1161,13 @@ message Channel {
     /// A set of flags showing the current state of the channel.
     string chan_status_flags = 19 [json_name = "chan_status_flags"];
+
+    /// The minimum satoshies this node is required to reserve in its balance.
```
`satoshis`?

roeierez replied:
Yeh.. fixed.

cfromknecht commented:
@roeierez latest change shows a linter error on the travis build, o/w think this is gtg

roeierez commented:
> @roeierez latest change shows a linter error on the travis build, o/w think this is gtg

fixed and also fixed my editor to run go fmt automatically.

cfromknecht commented:
@roeierez needs rebase since #3279 was merged

roeierez commented:
> @roeierez needs rebase since #3279 was merged

Rebased.

wpaulino commented on lnrpc/rpc.proto:
```diff
@@ -1161,6 +1161,13 @@ message Channel {
+    int64 local_chan_reserve_sat = 20 [json_name = "local_chan_reserve_sat"];
+
+    /// The minimum satoshis the other node is required to reserve in its 
+    /// balance.
```
Multi-line comments in the protos should use `/**/` instead.

roeierez replied:
Fixed.
